app/src/services/user/user.ts

```diff
@@ -1,89 +1,82 @@
+  onSuccess?: (data: any) => any;
+  onError?: (message: any) => any;
 }) => {
   try {
     const res = await axios(`/api/users/${userId}`);
-    let currentUserId;
-    if (userId === "me") {
-      const { id, avatar, firstName, lastName, jobTitle } = res.data;
-      setCurrentUserInfo(
-        JSON.stringify({ id, avatar, firstName, lastName, jobTitle })
-      );
-      currentUserId = id;
-    } else {
-      const currentUserInfo = await getCurrentUserInfo();
-      currentUserId = currentUserInfo?.id;
-    }
-    const {
-      firstName,
-      lastName,
-      jobTitle,
-      avatar,
-      id,
-    } = res.data as IUserRawResponse;
+
     const connectionIds = Object.keys(res.data.connections);
-    const connectionOfIds = Object.keys(res.data.connectionOf);
     const processedUserData: IUserProcessed = {
-      firstName,
-      lastName,
-      jobTitle,
-      avatar,
+      ...res.data,
       nOfConnections: connectionIds.length,
-      isAConnection: !!(
-        currentUserId && connectionOfIds.includes(currentUserId)
-      ),
-      id,
     };
-    onSuccess?.(processedUserData);
     return processedUserData;
   } catch (error) {
-    console.error(error);
-    typeof error?.message === "string" &&
-      onError("Unable to get info from server, please try again later");
+    if (error?.response?.status === 404) {
+      return {
+        id: userId,
+        ...User404,
+      };
+    } else {
+      throw error;
+    }
   }
 };
 
-const updateUser = async ({
-  data,
-  onSuccess,
-  onError,
-}: {
-  data: IUserPatchRequest;
-  onSuccess: (data: IUserPatchRequest) => void;
-  onError: (message: string) => void;
-}) => {
-  try {
-    const req = await axios({
-      method: "patch",
-      url: `/api/users/me`,
-      data,
-    });
-    if (req.status === 200) {
-      onSuccess(data);
-    } else {
-      onError(req.statusText);
+// TODO make more efficient route for getting multiple
+const getUsers = async ({ userIds }: { userIds: string[] }) => {
+  let currentUserId;
+  const res = await Promise.all(
+    userIds.filter((id) => id !== "me").map((userId) => getUser({ userId }))
+  );
+  const users: any = {};
+  if (userIds.includes("me")) {
+    const currentUser = await getUser({ userId: "me" });
+    if (!!currentUser) {
+      currentUserId = currentUser.id;
+      users[currentUserId] = currentUser;
     }
-  } catch (error) {
-    console.error(error);
-    typeof error?.message === "string" &&
-      onError(
-        "Sorry, we're unable to update your info at this time, please try again later"
-      );
+  }
+  res.forEach((user) => {
+    if (user) {
+      users[user.id] = user;
+    }
+  });
+  return {
+    users,
+    currentUserId,
+  };
+};
+
+const updateUser = async ({ data }: { data: IUserPatchRequest }) => {
+  const req = await axios({
+    method: "patch",
+    url: `/api/users/me`,
+    data,
+  });
+  if (req.status === 200) {
+    return data;
+  } else {
+    throw req.statusText;
```

Comment:
We're throwing an error object here. Should we not do
```suggestion
    throw new Error(req.statusText);
``` ?